### PR TITLE
Remove old packages to support Ruby 3.3

### PR DESCRIPTION
**Pull request by opal-storypark**
In order to upgrade to Ruby 3.3 we need to resolve the deprecation warnings caused by the `rfc822` package. This package is from 2014 and is not receiving updates. So I have chosen to remove the email functionality and the package as Storypark does not pass emails into Avatarly.

I've also removed the `unicode_utils`, it is a library from 2012 and doesn't seem to be required in modern ruby.

```console
$ bundle exec ruby --version && bundle exec rspec
ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [arm64-darwin23]
/Users/opalsymes/.gem/ruby/3.3.0/gems/fastimage-2.3.0/lib/fastimage.rb:62: warning: base64 was loaded from the standard library, but will no longer be part of the default gems since Ruby 3.4.0. Add base64 to your Gemfile or gemspec. Also contact author of fastimage-2.3.0 to add base64 into its gemspec.

Avatarly
  .generate_avatar
    generates avatar for given email address
    accepts parameters for size, background, vertical_offset and font colors
    accepts parameters for format
      .png
      .jpg
    non-email input
      uses first letters of first two space separated words
      falls back to dot-separated words when no spaces in input
      falls back to single-letter avatar when no dots and spaces found
      can generate using custom separators
      does not break if input has leading or trailing space
      does not break if input has a space and non-word character
      does not break if input has a dot and non-word character
      does not break if input has leading or trailing non-word character
      does not break if no text found
      strips leading/trailing whitespace without striping other whitespaces

Finished in 0.10105 seconds (files took 0.14536 seconds to load)
14 examples, 0 failures
```

### CODE REVIEW

File: avatarly.gemspec [hunk 1]
```diff
@@ -1,10 +1,8 @@
 Gem::Specification.new do |s|
   s.add_runtime_dependency('rmagick')
-  s.add_runtime_dependency('rfc822')
-  s.add_runtime_dependency('unicode_utils')
```
Comment (StoryparkNick) [inline avatarly.gemspec:4]: Good stuff, love to see some dependencies removed.

File: spec/avatarly_spec.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 describe Avatarly do
   describe '.generate_avatar' do
     it 'generates avatar for given email address' do
```
Comment (StoryparkNick) [inline spec/avatarly_spec.rb]: might be good to update the `generates avatar for given email address` to `for the provided name`

**Review by StoryparkNick — APPROVED** [review]
Nice one! Good to see this stuff simplified.

File: spec/avatarly_spec.rb (hunk 2 above)
Comment (opal-storypark) [reply spec/avatarly_spec.rb]:
```suggestion
    it 'for the provided name' do
```

**Pull request merged**